### ISSUE TITLE
[new feature request] - return all properties if user set full_properties=True

### ISSUE THREAD

**Issue by liucgbj** [zhmcclient/zhmc-ansible-modules#651]
i noticed that the `full_properties=True` have already been enabled by list method of console class of zhmcclient.
but in the following 4 modules, it seems to me the properties have been filtered, so would you please help to enhance it and return all properties if user set `full_properties=True` ?

```
zhmc_cpc_list.py
zhmc_password_rule_list.py
zhmc_user_list.py
zhmc_user_role_list.py
```

**Comment by andy-maier** [comment]
Currently, the Ansible list modules do not have a parameter to control whether all properties are returned. For performance reasons, I don't think we should always have the list modules return all properties, so we talk about a new module input parameter such as full_properties.

**Comment by andy-maier** [comment]
Needed for ZaaS (IAUT-613)

**Comment by andy-maier** [comment]
My proposal is to add two input parameters to each of the list modules:

* `additional_properties` - List of additional properties to be returned for each object, in addition to the default properties (see result description). The property names are specified with hyphens ('-'). This parameter is added only to resource types whose list operation supports the `additional-properties` query parameter, i.e. Partition, Adapter, VirtualSwitch. (Certificate does not have an Ansible module yet).

* `full_properties` - Boolean indicating that all properties of each object should be returned. This parameter is added to all list modules.

These two parms are mutually exclusive.

**Comment by andy-maier** [comment]
There is one caveat: The list modules are currently defined to return properties with underscores in their names (e.g. zhmc_adapter_list). When we get properties from the HMC either by full_properties or additional_properties, they have hyphens in their names. For backwards compatibility, we should translate these property names to have underscores, then the initial short list of properties is nicely expanded to a more complete list.

However, that then creates an inconsistency between returning facts on a single resource, where the HMC property names are currently returned.

We discussed that in the team, and agreed that we would use underscore names where possible, since that is better suitable for Ansible.

The current situation regarding property names is:

For single resource modules:
* input properties for resource creation or updates: contain underscores
* result properties: contain hyphens

For resource list modules:
* filter properties as input to list modules: contain underscores
* result properties: contain underscores

For the one exception where result properties of single resource modules currently are returned with hyphens, we can introduce an optional flag for requesting them to be returned with underscores, so that it remains compatible by default, but still allows users to consistently use underscores in property names in all situations.

**Comment by andy-maier** [comment]
@liucgbj PR #744 is ready for review.

**Comment by liucgbj** [comment]
Got it, I will arrange someone from dev to follow up with you on it, thank you.

**Comment by andy-maier** [comment]
PR #744 has been merged and will be in the upcoming version 1.7.0
